docs/commands/build.md

```diff
@@ -20,6 +21,92 @@ this-is-param-case) in `dist/`
 
 ## Options:
 
-`--private`: Allows the building of private packages
+`--private`: Allows the building of private packages.
 
-`--preserve-modules`: Preserve module structure in generated modules
+`--preserve-modules`: Preserve module structure in generated modules.
+
+`--changed`: Build only packages whose workspaces contain files that have
+changed. Files that have changed are calculated comparing the current state of
+the repository with the branch specified by `compareBranch` or, if
+`compareBranch` is not set, with the default git branch.
+
+`--compareBranch`: Specify the comparison branch used to determine which files
+have changed when using the `changed` option. If this option is used without
+`changed`, the command will fail.
+
+`--descendants`: Build the packages specified by the `[packages...]` argument
+and/or the `--changed` option and additionally build all their descendants (i.e.
+the packages that have a direct or indirect dependency on them) in dependency
+order.
+
+`--ancestors`: Build the packages specified by the `[packages...]` argument
+and/or the `--changed` option and additionally build all their ancestors (i.e.
+the packages that have a direct or indirect dependency on them) in dependency
+order.
+
+## Dependency selection and build order examples
+
+We'll be using this configuration of internal (workspace) dependencies in the
+following examples.
+
+```js
+{
```

Comment:
Is this example a bit inaccessible to others?

Maybe we should just say `dependencies` instead of `workspaceDependencies`, and throw in a small handul of other things like `react` for example, even though they're not relevant, just to make it a bit closer to a real example.

Reply:
I added extracts of all the `package.json`s in our imaginary monorepo and mentioned that they were resolved internally to the data structure above (which is easier to read).